### PR TITLE
[Chore] Add compatibility e2e test

### PR DESCRIPTION
**Pull request by IshwarKanse**
Adds e2e test case for checking compatibility of Tempo Operator with other projects by probing the endpoints using curl, or verifying the port names in the manifests etc. - without installing the other application, just by simulating the API requests, i.e. checking the "API contract" between the Tempo operator and the other product.

Currently this test case checks compatibility with Grafana but we can extend it to add checks for other projects like OpenShift Service Mesh.

### CODE REVIEW

**Comment by IshwarKanse** [comment]
This passed on both OpenShift and Kind cluster tests locally. Does the CI doesn't have enough resources ?

File: tests/e2e/compatibility/01-install.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  namespace: kuttl-compatibility
+spec:
+  storage:
+    secret:
+      name: minio
+      type: s3
+  storageSize: 200M
+  resources:
+    total:
+      limits:
```
[the remark this answers is not in the document]
Comment (IshwarKanse) [reply tests/e2e/compatibility/01-install.yaml]: I removed the limits and the test ran fine.

**Comment by andreasgerstmayr** [comment]
Except for the last step, this e2e test is identical to `smoketest-with-jaeger`, right?

I wonder if we should just add the last step from here to the `smoketest-with-jaeger` (and possibly rename `smoketest-with-jaeger`)? This would avoid duplication and reduce the time to run our e2e test suite, what do you think?

**Comment by IshwarKanse** [comment]
@andreasgerstmayr Sounds good to me, I'll update the PR tomorrow and rename smoketest-with-jaeger.

**Comment by IshwarKanse** [comment]
@andreasgerstmayr Updated the PR.

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**